backend/backend/celery_script.py

```diff
@@ -0,0 +1,189 @@
+import requests
+from bs4 import BeautifulSoup
+from datetime import datetime
+import re
+from news.models.article import Article
+
+
+class ArticleScraper:
+    def __init__(self, start_url, target_date=None):
+        self.start_url = start_url
+        self.target_date = target_date or datetime.today().strftime('%B %d, %Y')
+
+    @staticmethod
+    def convert_to_date(date_string):
+        date_formats = [
+            "%B %d, %Y",  # e.g., August 21, 2024
+            "%d %B %Y",  # e.g., 21 August 2024
+            "%Y-%m-%d"  # e.g., 2024-08-21
+        ]
+        for fmt in date_formats:
+            try:
+                date_obj = datetime.strptime(date_string, fmt)
+                return date_obj.strftime("%Y-%m-%d")
+            except ValueError:
+                continue
+        raise ValueError(f"Date format for '{date_string}' is not recognized.")
+
+    @staticmethod
+    def fetch_and_parse(url):
+        try:
+            response = requests.get(url)
+            response.raise_for_status()
+            return BeautifulSoup(response.content, 'html.parser')
+        except requests.exceptions.RequestException as e:
+            print(f"Error fetching the URL: {url}\nError: {e}")
+            return None
+
+    @staticmethod
+    def extract_text_from_tag(tag, default_text='No data'):
+        return tag.text.strip() if tag else default_text
+
+    @staticmethod
+    def extract_post_date(post_soup):
+        detail_left_tittle = post_soup.find('div', class_='detail-left-tittle')
+        if detail_left_tittle:
+            date_tag = detail_left_tittle.find('div', class_='category-date')
+            return ArticleScraper.extract_text_from_tag(date_tag, 'No date')
+        return 'No date'
+
+    @staticmethod
+    def extract_image_link(post_soup):
+        image_div = post_soup.find('div', class_='medium-insert-images')
+        if not image_div:
+            return 'No image'
+        image_tag = image_div.find('img')
+        return image_tag['src'] if image_tag else 'No image'
+
+    @staticmethod
+    def extract_metadata(post_soup):
+        metadata = {}
+        detail_heading = post_soup.find('div', class_='detail-heading')
+        if detail_heading:
+            metadata['post_title'] = ArticleScraper.extract_text_from_tag(detail_heading.find('h1'), 'No title')
+        else:
+            metadata['post_title'] = 'No title'
```

Comment:
There were some of the pages/links in the website having no heading/author-name and published date, and by inspecting it, i came to know that these links are for live broadcasts, so we are ignoring them, As far as articles are concerned we have all available data for each field, but made these `No title` checks for edge cases. And if any of such case occur that we are not getting anything and causing any exception in the code, we are passing that exception because don't want out code to raise exceptions.